parse/lex_test.go

```diff
@@ -550,21 +578,38 @@ func assertItems(t *testing.T, want []item, l *lexer) {
 
 	logItems := make([]func(), 0, len(want))
 
-	for _, wantItem := range want {
-		got := l.nextItem()
+	got := make([]item, 0, len(want))
+
+	for {
+		itm := l.nextItem()
+		got = append(got, itm)
+
+		if itm.typ == itemError || itm.typ == itemEOF {
+			break
+		}
+	}
+
+	if len(got) != len(want) {
+		t.Errorf(`want %d items, got %d`, len(want), len(got))
+
+		return
```

Comment:
If this fails, it is hard to understand why exactly - no visibility of items.

Can we simplify by using `slices.Equal()` now?